include/fmt/format-inl.h

```diff
@@ -181,13 +181,21 @@ void format_error_code(internal::buffer& out, int error_code,
   assert(out.size() <= inline_buffer_size);
 }
 
+// try an fwrite, FMT_THROW on failure
+void fwrite_fully(const void *ptr, size_t size, size_t count, FILE *stream) {
+  size_t written = std::fwrite(ptr, size, count, stream);
+  if (written < count) {
+    FMT_THROW(system_error(errno, "cannot write to file"));
+  }
+}
+
 void report_error(FormatFunc func, int error_code,
                   string_view message) FMT_NOEXCEPT {
   memory_buffer full_message;
   func(full_message, error_code, message);
   // Use Writer::data instead of Writer::c_str to avoid potential memory
   // allocation.
-  std::fwrite(full_message.data(), full_message.size(), 1, stderr);
+  fwrite_fully(full_message.data(), 1, full_message.size(), stderr);
```

Comment:
Why swap the order of arguments?

Reply:
Oh, it's not important but it looked like a mistake to me. There are 3 fwrite calls.
The other two calls put the char size first.

    fwrite(s.data, 1, s.size, fp)
    fwrite(ws.data, sizeof(wchar_t), ws.size, fp)

... which match fwrite's intended semantics of having the record_size followed by the record_count.

Reply:
I disagree.  The new way tells the system to write COUNT records of 1 byte each.  Conceptually I think it's more correct to say we want to write 1 record of COUNT bytes, which is what the old code did.  Although I'm quite positive that virtually every real implementation simply multiplies these numbers together, one could imagine an implementation where the new change would have the underlying system call write() invoked COUNT times with a size of 1 byte, which would be super inefficient, whereas the original code would invoke the underlying system call 1 time with COUNT bytes.